charts/istio-alerts/templates/service-prometheusrule.yaml

```diff
@@ -85,5 +82,36 @@ spec:
         {{ toYaml . | nindent 8}}
         {{- end }}
     {{- end }}
-  {{- end }}
+    {{- end }}
+    {{- with .Values.serviceRules.destinationServiceSelectorValidity }}
+    {{- if .enabled }}
+    - alert: DestinationServiceSelectorValidity
+      annotations:
+        summary: >-
+          DestinationServiceSelector for istio-alerts is invalid
+        runbook_url: {{ $values.defaults.runbookUrl}}#Alert-Rules-Selectors-Validity
+        description: >-
+          The DestinationServiceSelector used for service level alerts did not return any data.
+          Please check the DestinationServiceSelector applied in your istio-alerts chart
+          is correct to ensure you are properly selecting your service so that you
+          will be alerted for service issues. The current selector is
+          `{destination_service_namespace="{{ $release.Namespace }}", destination_service_name="{{ $.Values.serviceRules.destinationServiceName }}"}`.
+      expr: >-
+        (
+          count(
+            kube_service_info{
+              namespace={{ $.Release.Namespace }},
+              name={{ $.Values.serviceRules.destinationServiceName }}
+            }
+          ) or on() vector(0)
```

Comment:
So this part of the alert is a little tricky ... if you're missing the metric, it shows `0`:
![image](https://github.com/Nextdoor/k8s-charts/assets/768067/72b12345-027d-42fa-b68a-6298e6dc3f99)

However, if we typed the metric name wrong, it also reports `0`.. (or as you pointed out, if you are missing kube-state-metrics):
![image](https://github.com/Nextdoor/k8s-charts/assets/768067/8c68a7af-a322-4605-97fb-63f3efcbd0a6)

Perhaps we just want to add a second rule in here (thats under the same condition) that verifies that kube-state-metrics is working too and if its not, sends a separate alarm with useful info to tell us that we broke something related to the kube-state-metrics?

```
(count(up{job="kube-state-metrics"}) or on() vector(0) == 0)
```

or

```
(count(kube_service_info{}) or on() vector(0) == 0)
```

Reply:
I think in order to avoid scope creep, in this chart we have to assume KSM is working. In the README, KSM is now listed as a requirement and there is a flag to disable this alert if you do not want to install KSM.

Reply:
KSM health tracking should be done in a KSM related chart. It would not make sense for all KSM dependant charts to have independant/redundant KSM health alerts in place IMO.

Reply:
I'll accept that for now ... I reserve the right to come back to this discussion. :P